[Ethan Khor] iP

squeakysquak opened the pull request:
### **DukeyJukey**
> “Use the force Luke.” – Gandalf, Harry Potter

DukeyJukey will help you keep track of tasks. It is:
- easy to learn
- Fast to use

Just follow these steps:

1. Download [here](https://youtu.be/yLuvK4zN0Yw)
2. Open it
3. Add your tasks
4. 😉

Features:
- [X] Managing tasks
- [X] GUI

This is a `main` method

```
public class Main {
    public static void main(String[] args) {
        System.out.print("Sparkle up your day (TM).");
    }
}
```

Commit:
Add Gradle support

squeakysquak committed:
Level-1

squeakysquak committed:
Level-2

squeakysquak committed:
Level-3

squeakysquak committed:
added ToDo,Deadline and Event subclasses

squeakysquak committed:
Level-4

squeakysquak committed:
Level-5

squeakysquak committed:
Level-6

squeakysquak committed:
A-TextUiTesting

squeakysquak committed:
separate classes into their own files

squeakysquak committed:
add file functionality for list, todo, deadline and event

squeakysquak committed:
add file functionality for done and delete

squeakysquak committed:
add LocalDate support to Event and Deadline class

squeakysquak committed:
add duke package

squeakysquak committed:
Merge branch 'master' into branch-Level-7

squeakysquak committed:
Merge branch 'master' into branch-Level-8

# Conflicts:
#	src/main/java/duke/Deadline.java
#	src/main/java/duke/Event.java

squeakysquak committed:
Merge branch 'branch-Level-7'

squeakysquak committed:
Merge branch 'branch-Level-8'

squeakysquak committed:
update deadline, event and todo to use their classes, include a remove command which does the same thing as delete

squeakysquak committed:
add new classes

squeakysquak committed:
implement ui, storage and tasklist

squeakysquak committed:
implement parser

squeakysquak committed:
implement JUnit tests

squeakysquak committed:
make jar file

squeakysquak committed:
add Javadoc comments

squeakysquak committed:
fix formatting

squeakysquak committed:
implement find function

squeakysquak committed:
change some Ui messages

squeakysquak committed:
add some more comments

squeakysquak committed:
Merge branch 'branch-A-JavaDoc'

squeakysquak committed:
change the display message for searching

squeakysquak committed:
Merge branch 'branch-Level-9'

squeakysquak committed:
Merge branch 'master' into branch-A-CodingStandard

squeakysquak committed:
fix some formatting in Ui

squeakysquak committed:
Merge branch 'branch-A-CodingStandard'

squeakysquak committed:
Merge branch 'add-gradle-support' of https://github.com/squeakysquak/ip and add Launcher class for GUI

squeakysquak committed:
add GUI

squeakysquak committed:
fix formatting

squeakysquak committed:
Merge branch 'branch-Level-10'

itzblim commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,172 @@
+                ui.sayBye();
+                break;
+            } else if (command.equals("list")) {
+
```
Perhaps the spaces here could be removed, if they're not needed that is? I'm not sure what checkstyle's response is!

itzblim commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,75 @@
+ * Represents Duke's storage system and handles file operations.
+ */
+public class Storage {
+    //private static List<String> lines;
```
I'm guessing you intend to remove this in the end? Since it's not really a comment that describes the code but probably an intermediate step you took?

itzblim commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,63 @@
+ */
+public class Task {
+    String name;
+    boolean done;
```
Would it be appropriate to perhaps make these variables private, if you will be providing setters anyways?

itzblim commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,89 @@
+    private Integer month;
+    private Integer day;
+
+    public Deadline(String name, String date) {
```
Would it be helpful to provide a javadoc comment for this public constructor?

itzblim commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,47 @@
+        this.getChildren().setAll(tmp);
+    }
+
+    public static DialogBox getUserDialog(Label l, ImageView iv) {
```
Same goes for here with regards to the javadoc comments for public methods, if you would like to?

itzblim commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,316 @@
+    /**
+     * Gets the response to a command from Duke.
+     *
+     * @param command The command that was entered.
```
I believe you might want to include an @return here?

itzblim reviewed (commented):
Overall the coding standard is carefully followed in most of the code! Just a few small places to review! 👍

JoelChanZhiYang commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,89 @@
+            monthString = "December";
+        } else {
+            throw (new IllegalArgumentException("uh oh stinky"));
+        }
```
Perhaps here would be a good place to use a switch statement.

JoelChanZhiYang commented on resources/view/DialogBox.fxml:
```diff
@@ -0,0 +1,20 @@
+        <Label fx:id="dialog" text="Label" textFill="WHITE" wrapText="true">
+         <font>
+            <Font name="Courier New Bold" size="20.0" />
+         </font></Label>
```
I believe the `</Label>` should be in the next line.

JoelChanZhiYang commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,316 @@
+        userInput.setOnAction((event) -> {
+            handleUserInput();
+        });
+    }
```
You could add abstract out these attributes into a fxml file.

JoelChanZhiYang commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,88 @@
+            monthString = "December";
+        } else {
+            throw (new IllegalArgumentException("uh oh stinky"));
+        }
```
This looks like a good place to use a switch statement

JoelChanZhiYang commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,172 @@
+                }
+                taskNo--;
+                taskList.makeDone(storage, taskNo);
+            } else if (command.contains("delete") || command.contains("remove")) {
```
Will using `contains` lead to some logic bugs? For example, what if the user types in the command 
`todo Delete unwanted pictures from gallery`

JoelChanZhiYang commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,176 @@
+/**
+ * Represents the user interface and displays messages for Duke.
+ */
+public class Ui {
```
Since this class does not have any state, would it more appropriate for it's methods to be static?

JoelChanZhiYang reviewed (commented):
Overall, very readable code!

squeakysquak committed:
add assertions for wipe, list and find commands.

squeakysquak committed:
Merge branch 'branch-A-Assertions'

squeakysquak committed:
improve readability

squeakysquak committed:
Merge branch 'branch-A-CodeQuality'

squeakysquak committed:
add javadoc comments to new extracted methods and improve javadoc comments for TaskListInternal

squeakysquak committed:
Merge branch 'branch-A-CodeQuality'

squeakysquak committed:
implement friendlier syntax and bug fixes

squeakysquak committed:
Merge branch 'branch-C-FriendlierSyntax'

squeakysquak committed:
Add JavaFX dependencies to build.gradle file

Gradle can bundle the JavaFX third-party library into the jar file. If
JavaFX is not bundled with the JAR file, the application will not work
in computers that don't have JavaFX installed.

As a step towards preventing this, let's add JavaFX dependencies
to the build.gradle file. Doing so will allow us to bundle the
JavaFX third-party library into the jar file.

squeakysquak committed:
add multiple facial expressions for Duke and a welcome message.

squeakysquak committed:
fix a bug and make userguide

squeakysquak committed:
improve code quality

squeakysquak committed:
Set theme jekyll-theme-cayman

squeakysquak committed:
Update README.md

squeakysquak committed:
update README.md

squeakysquak committed:
fix Ui.png

squeakysquak committed:
update README.md and fix Ui.png again

squeakysquak committed:
remove Ui.png

squeakysquak committed:
fix and add Ui.png

squeakysquak committed:
fix build.gradle

squeakysquak committed:
fix file problem, fix bye command to exit program

squeakysquak committed:
update UG

squeakysquak committed:
update UG

squeakysquak committed:
update UG
